tests/lib/ignored-paths.js

```diff
@@ -529,50 +529,74 @@ describe("IgnoredPaths", function() {
             var ignoredPaths = new IgnoredPaths({ ignore: true, ignorePath: getFixturePath(".eslintignoreWithUnignoredDefaults"), cwd: getFixturePath() });
 
-            var actual = ignoredPaths.getIgnoredFoldersGlobPatterns();
+            var shouldIgnore = ignoredPaths.getIgnoredFoldersGlobChecker();
 
-            assert.notInclude(actual, "node_modules/**");
-            assert.notInclude(actual, "bower_components/**");
+            assert.isTrue(shouldIgnore("node_modules/a"));
+            assert.isTrue(shouldIgnore("node_modules/a/b"));
+            assert.isTrue(shouldIgnore("bower_components/a"));
+            assert.isTrue(shouldIgnore("bower_components/a/b"));
+            assert.isFalse(shouldIgnore(".hidden"));
+
+            assert.isFalse(shouldIgnore("node_modules/package"));
+            assert.isFalse(shouldIgnore("bower_components/package"));
         });
 
-        it("should return dirs with unignored defaults in ignore file when ignore option is disabled", function() {
+        it("should ignore files which we try to re-include in ignore file when ignore option is disabled", function() {
             var ignoredPaths = new IgnoredPaths({ ignore: false, ignorePath: getFixturePath(".eslintignoreWithUnignoredDefaults"), cwd: getFixturePath() });
 
-            var expected = ["node_modules/**", "bower_components/**"];
-            var actual = ignoredPaths.getIgnoredFoldersGlobPatterns();
+            var shouldIgnore = ignoredPaths.getIgnoredFoldersGlobChecker();
+
+            assert.isTrue(shouldIgnore("node_modules/a"));
+            assert.isTrue(shouldIgnore("node_modules/a/b"));
+            assert.isTrue(shouldIgnore("bower_components/a"));
+            assert.isTrue(shouldIgnore("bower_components/a/b"));
+            assert.isFalse(shouldIgnore(".hidden"));
 
-            assert.includeMembers(actual, expected);
+            assert.isTrue(shouldIgnore("node_modules/package"));
+            assert.isTrue(shouldIgnore("bower_components/package"));
         });
 
-        it("should not return dirs unignored by ignorePattern", function() {
+        it("should not ignore dirs which are reincluded by ignorePattern", function() {
             var ignoredPaths = new IgnoredPaths({ ignore: true, cwd: getFixturePath("no-ignore-file"), ignorePattern: "!/node_modules/package" });
 
-            var actual = ignoredPaths.getIgnoredFoldersGlobPatterns();
+            var shouldIgnore = ignoredPaths.getIgnoredFoldersGlobChecker();
+
+            assert.isTrue(shouldIgnore("node_modules/a"));
+            assert.isTrue(shouldIgnore("node_modules/a/b"));
+            assert.isTrue(shouldIgnore("bower_components/a"));
+            assert.isTrue(shouldIgnore("bower_components/a/b"));
+            assert.isFalse(shouldIgnore(".hidden"));
 
-            assert.notInclude(actual, "node_modules/**");
-            assert.include(actual, "bower_components/**");
```

Comment:
Actually, @alberto [added](https://github.com/eslint/eslint/commit/064e298d84f600d55e803a28c4f945b42a62f2fa) a new "private" method `ignoredPaths.getIgnoredFoldersGlobPatterns()`  along with there test cases to temporarily fix the perf issue.

`ignoredPaths.getIgnoredFoldersGlobPatterns()` is only used to fetch the glob patterns for `options.ignore` of glob. And this pr changes the behavior of this method, as well as unit tests.